src/components/ConfirmDialog.jsx

```diff
@@ -0,0 +1,113 @@
+import './ConfirmDialog.css';
+import React from 'react';
+import {
+	MaterialButton,
+	buttonStyle,
+	whiteFontColor,
+} from './material-buttons';
+import {
+	Typography,
+	IconButton,
+	Slide,
+	Dialog,
+	DialogTitle,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+
+// MUI's Dialog already comes with built-in focus management and accessibility features.
+// It automatically traps focus inside the modal, moves focus to the modal when it opens,
+// and returns it to the previously focused element when it closes.
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+	return <Slide direction="left" ref={ref} {...props} />;
+});
+
+const dialogStyle = {
+	backgroundColor: 'rgb(20, 20, 20, 0.8)',
+	position: 'absolute',
+	top: 20,
+	right: 20,
+};
+
+const typographyStyle = {
+	padding: '1em',
+	...whiteFontColor,
+};
+
+export function ConfirmDialog({ props }) {
+	const { handleDelete, title, setOpen, open } = props;
+
+	const handleClose = (event) => {
+		setOpen(false);
+		event.target.type === 'submit' && handleDelete();
+	};
+
+	return (
+		<Dialog
+			open={open}
+			TransitionComponent={Transition}
+			onClose={handleClose}
+			PaperProps={{
+				style: dialogStyle,
+			}}
+			BackdropProps={{
+				invisible: true,
+			}}
+			sx={{
+				'& .MuiDialog-container': {
+					justifyContent: 'flex-end',
+					alignItems: 'flex-start',
+				},
```

Comment:
Group PR review: consider saving this prop object as a `const` variable and then passing it in here to match the other props passed in.

Thoughts are this would make it easier to edit or update because you'd be editing a variable rather than needing to find an inline object on a prop passed into `Dialog`.

Reply:
Good example https://github.com/the-collab-lab/tcl-75-smart-shopping-list/pull/72/files#r1789247967